### DISCUSSION TITLE
Running WindNinja through a Cloud Service?

### DISCUSSION THREAD

**Discussion by DanielSWx** [firelab/windninja#518] [General]
Hi. I am currently running WindNinja through my local computer, but I wanted to ask about the feasibility of running the program through a cloud service such as Google Cloud or AWS. Is it feasible/possible to do so, or would it require something else that would necessitate changing the application? I just wanted to ask prior to moving forward; if it's not possible, I completely understand! Thank you!

**Comment by nwagenbrenner** [comment]
Hi Daniel,

Yes, running WindNinja through a cloud service is totally possible. In
fact, we do this for our WindNinja Mobile app. We have the backend running
the WindNinja command line interface in an Azure environment (we used to
use AWS too). We have a flask server set up to accept requests from
the phone, set up and fire off WindNinja simulations, and the package and
send the needed outputs to the phone for viewing. Hope this helps. Feel
free to reach out if you have additional questions.

Thanks,
Natalie

**Reply by JEFlynn** [reply]
Hi Natalie! How exactly is this accomplished? I'm trying to get a docker version of Windninja running up on AWS, following the linux guide found here: https://github.com/firelab/windninja/wiki/Building-WindNinja-on-Linux but I'm hitting all sorts of walls. 

I modified the referenced script to work with the AWS linux distro (edit: it looks like specifically I need to do it on Amazon Linux 2023. Which is going to be even weirder. I was trying on Their Amazon 2 distro), but I keep hitting major roadblocks with the cmake step. If a modern installed version of boost is used, it throws all sorts of conflict errors about certain boost functions. If I try to grab old versions of boost, that seems to be failing too. 

The setup seems to want boost 1.41.1 which I can't find copies of. 1.41.0 fails and 1.42.0 seems to have removed certain functions that something is attempting to call.

I'm very stuck. 

Final Edit: Actually, scratch that. I got everything to build after a lot of trial and error. Most of the commands didn't cooperate in a script so I just ran them by hand or hardcoded some env values. 

However, everything ran and I was able to do the make  and make install commands, but I'm not sure what's next. It says the CLI is now in the cli folder and that WindNinja should be a command now, but that doesnt seem to be the case. And trying to call WindNinja_cli says that it errors while loading shared libraries. I feel like I'm so close!

**Comment by DanielSWx** [comment]
Great, thank you so much! I sincerely appreciate this!
